builder/comp-builder.nix

```diff
@@ -404,7 +413,15 @@ let
       (lib.optionalString stdenv.hostPlatform.isWindows ''
         export pkgsHostTargetAsString="''${pkgsHostTarget[@]}"
       '') +
-      (if stdenv.hostPlatform.isGhcjs then ''
+      # this could be refactored but would lead to many rebuilds
+      (if CAComponent then ''
+        runHook preBuild
+        # we need avoid parallel building (passing -j1) in order to have a deterministic output and therefore
+        # avoid potential situations where the binary cache becomes useless
+        # https://gitlab.haskell.org/ghc/ghc/-/issues/12935
```

Comment:
Can we please make this a nix comment (that way if we expand it or modify it later it does not require rebuilds)?

Reply:
Oops, fixed